app/models/spree/variant.rb

```diff
@@ -297,5 +298,11 @@ def destruction
       exchange_variants(:reload).destroy_all
       yield
     end
+
+    def ensure_unit_value
+      return unless product&.variant_unit == "items" && unit_value.nil?
+
+      self.unit_value = 1.0
```

Comment:
If the issue is that we rely on `unit_value` even when `product.variant_unit == "items"` we should extend the validation at line 58, right?

Reply:
I think it's slightly different what we need to do; in the case of that validation, we're saying if you try to save a variant with `weight` or `volume`, **you** must provide the `unit_value` or it'll be invalid. 

Here, we're more saying that if you try to save a variant with `items` and don't provide a `unit_value`, because the UI doesn't really make it clear that you need to, or allow you to, then we're going to set it to be 1.0 for you.

Reply:
Yeah, and I think in the actual cases that are causing these issues, the user doesn't have a chance to update that field. Like if the product is changed to `items` in the bulk products page, it's not possible to adjust this other field on the variants at the same time.